include/ddc/kernels/splines/matrix_sparse.hpp

```diff
@@ -122,17 +122,20 @@ class Matrix_Sparse : public Matrix
         std::shared_ptr const gko_exec = create_gko_exec<ExecSpace>();
         m_matrix_dense = gko::matrix::Dense<
                 double>::create(gko_exec->get_master(), gko::dim<2>(mat_size, mat_size));
-        m_matrix_dense->fill(0);
         m_matrix_sparse = matrix_sparse_type::create(gko_exec, gko::dim<2>(mat_size, mat_size));
+
+        m_matrix_dense->fill(0);
+        m_matrix_sparse->clear();
```

Comment:
```suggestion
        m_matrix_sparse->clear();
```

Can you explain what it does and why it is required ? Looking at the doc https://ginkgo-project.github.io/ginkgo-generated-documentation/doc/develop/classgko_1_1PolymorphicObject.html#a8bb6f9995e8166099ca3ecab51d98df5 I am not sure we need to call it just after the object has been created